### ISSUE TITLE
Camera displays in vehicles can show security camera views as night time even though it is daylight in some circumstances.

### ISSUE THREAD

**Issue by Relgar** [cataclysmbn/Cataclysm-BN#1196]
# Describe the bug
Not quite sure under what circumstances it occurs, but when viewing through a security camera system sometimes, the camera display will show the security camera view as though it were night time even though it is daylight outside.

# Steps To Reproduce
1. Download the uploaded save.
2. Turn on camera display system located at the seat in the vehicle (player starts here), cameras are located at the 4 corners of the vehicle, giving a 360 degree view.
3. Observe the night view.
4. Open hatch doors on either side.
5. It is morning

# Expected behavior
1. Morning light view through the camera system

# Screenshots

![image](https://user-images.githubusercontent.com/84274761/147349538-11e07ab0-af59-44a9-ac76-5d81cee39c43.png)

![image](https://user-images.githubusercontent.com/84274761/147349567-00386ffd-cf53-427f-95c9-769838b4797a.png)

# Versions and configuration
- OS: Windows 10 Home
    - OS Version: 10.0.19044
- Game Version: 24952e2 [64-bit]
- Graphics Version: Tiles
- Game Language:  []
- Mods loaded: [
    Bright Nights [dda],
    Disable NPC Needs [no_npc_food],
    Simplified Nutrition [novitamins],
    Aftershock [aftershock],
    Hauling Space From Aftershock [aftershock_hauling_space],
    BioCo Content [bioco],
    Craft Ammo Mags & Belt Links [SD_Magcrafting],
    Culinary Days Ahead [culinary_days_ahead],
    Expanded Archery [ExArch],
    Gourmand's Graces [Gourmand'sGraces],
    Homestead Instruments Inc. [homestead],
    Make a Lot of Items [Make_a_Lot_of_Items_Mod],
    Medieval Mod Reborn [Medieval_Mod_Reborn],
    Modern Weapon Pack Expanded [modern_weapon],
    More Survivor Notes [more_survive_notes],
    MST Extra [MST_Extra],
    Sewing Machine [SM],
    The Glory Of Cereal! [CerealMod],
    Draco's Dog Mod [DracoDogMod],
    Draco's Pig Mod [DracoPigMod],
    Modular Turrets with No Blacklist [modular_turrets_no_blacklist],
    Salvaged Robots* [Salvaged_Robots__],
    Convenient Constructions [convenientconstructions],
    Food Irradiator [food_irradiator],
    Mining Mod [Mining_Mod],
    More Survivor Stuff [more_survivor_stuff],
    Mythical Martial Arts [MMA],
    NPC Recruitment Options [recruitment_options],
    Wandering Masters [wandering_masters],
    Alternative Basement Layouts [basement_extension],
    More Locations* [more_locations__],
    Ships From Oa Additional Buildings [Ships],
    Trailer Parks [trailerpark],
    Whaleys TALL Buildings [whaleys_tallbuild],
    Tankmod: Revived [Tankmod_Revived],
    Artyoms' Gun Emporium - Reloaded [Arts' Guns],
    Targeted Mutations [TaMu],
    Autodoc Harmless and Perfect [autodoc_harmless_perfect],
    Bionic Slots [cbm_slots],
    Engineering essentials [EngineeringEssentials],
    Fuji's Military Profession Pack [fuji_mpp],
    Manual Bionic Installation [manualbionicinstall],
    No Addictions [No_Addictions],
    No Bionic Slots [no_cbm_slots],
    No Filthy Clothing [no_filthy_clothing],
    No Lifting Reqs [disable_lifting],
    Nonperishable Overhaul [Nonperishable_Overhaul],
    Prevent Zombie Revivication [no_reviving_zombies],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills],
    THRESHOLD [add_mutant],
    Aftershock - Easydocs [AFS_Easydocs]
]

# Additional context
Save File
[Camera Display.zip](https://github.com/cataclysmbnteam/Cataclysm-BN/files/7774182/Camera.Display.zip)

**Comment by Coolthulhu** [comment]
What do you mean by "night view"?

Cameras have a range offset, which means that they see things as if they were further away. I think it's 20 tiles for most cameras.
With the hatches open, it looks like distant tiles are dark. This is because perceived light of a tile decreases with distance.
The most likely explanation I see here is that the camera range offset is making them see things as if they were still in that "dark range". As the day becomes brighter, the "day range" would expand and it should look like normal.

**Comment by Relgar** [comment]
By night view I mean it looks as though it is still night time when view through the camera. Let me run through this save a little more and see if the camera view changes as the time solders on - either way, it does sound like the offset thing, but it was just very disconcerting that looking through a camera would show 'dim light' when it's actually bright outside.

**Comment by Relgar** [comment]
Yup, vision in the cameras returns to normal as the day progresses. Should this 20 tile offset be removed? What was the original intent with the inclusion?

**Comment by Coolthulhu** [comment]
> What was the original intent with the inclusion?

Cameras having worse range than eyes.

**Comment by Relgar** [comment]
Is it possible to cut off max range while preserving the ratio of dim-to-bright tiles? I.e. if max eyesight range has a radius of say 20 bright with 2 dark tiles, could cameras be set to 10 bright / 1 dim? Whatever they currently have implemented feels off by a good amount, hence this weird edge case scenario.
